k8s-rule.yaml

```diff
@@ -0,0 +1,1034 @@
+- message: Deprecated/removed Kubernetes API version 'extensions/v1beta1' is used for 'Deployment'. Consider using 'apps/v1'.
+  ruleID: k8s-deprecated-api-001
+  description: "Check for usage of deprecated Kubernetes API versions"
+  category: potential
+  effort: 2
+  when:
+    yaml.k8sResourceMatched:
+        apiVersion: "extensions/v1beta1"
+        kind: "Deployment"
+        deprecatedIn: "v1.9.0"
```

Comment:
What is the need for passing deprecated / removed versions as parameters here? IMO `k8sResourceMatched` should only care about matching on given fields in a resource and should have nothing to do with the deprecated versions etc. The rule author should use source / target labels to convey this information. Am I missing something?

Reply:
@pranavgaikwad yes in your approach we might don't need those parameters but during the project implementation we thought if it's possible to get the target version of k8s then users don't need to pick rules manually they can use the full rule-set for evaluation and application should create the violation upon the actual deprecated or removed api-versions according to target version .

Reply:
@pranavgaikwad Given that we have #466, I am wondering if we can work on that issue as a follow-up and merge this as is because main is tracking 0.4.0 now?